apps/condo/domains/property/components/panels/Builder/MapConstructor.ts

```diff
@@ -0,0 +1,490 @@
+            .map(section => section.floors
+                .map(floor => floor.index))
+            .flat()
+        )
+    }
+    
+    get minFloor (): number {
+        return Math.min(...this.map.sections
+            .map(section => section.floors
+                .map(floor => floor.index))
+            .flat()
+        )
+    }    
+
+    get possibleFloors (): number[] {
+        const floors = []
+        for (let floorIndex = this.minFloor; floorIndex <= this.maxFloor; floorIndex++) {
+            if (floorIndex !== 0) {
+                floors.unshift(floorIndex)
+            }
+        }
+        return floors
+    }
+
+    get isEmpty (): boolean {
+        return this.map.sections.length === 0
+    }
+
+    get sections (): BuildingSection[] {
+        return this.map.sections
+    }
+    
+    protected getUnitIndex (unitId: string): IndexLocation {
+        const result = { section: -1, floor: -1, unit: -1 }
+        try {
+            this.map.sections.forEach((section, sectionIdx) => {
+                section.floors.forEach((floor, floorIdx) => {
+                    const unitIdx = floor.units.findIndex(unit => unit.id === unitId)
+                    if (unitIdx !== -1) {
+                        result.section = sectionIdx
+                        result.floor = floorIdx
+                        result.unit = unitIdx
+                        throw new Error('Found')
+                    }
+                })
+            })
+        } catch (_) {
+            //
+        }
+        return result
+    }
+
+    public getUnitInfo (id: string): BuildingUnit {
+        const newUnit = { id: null, type: 'unit', label: '', floor: '', section: '' }
+        if (!id) {
+            return newUnit
+        }
+        const unitIndex = this.getUnitIndex(id)
+        if (unitIndex.unit === -1) {
+            return newUnit
+        }
+        const { label, type } = this.map.sections[unitIndex.section].floors[unitIndex.floor].units[unitIndex.unit]
+        return {
+            id,
+            label, 
+            type,
+            section: this.map.sections[unitIndex.section].id,
+            floor: this.map.sections[unitIndex.section].floors[unitIndex.floor].id,
+        }
+    }
+
+    public  getSectionOptions (): BuildingSelectOption[] {
+        const options = this.sections.map(section => ({ id: section.id, label: section.name }))
+        return options
+    }
+    
+    public getSectionFloorOptions (id: string): BuildingSelectOption[] {
+        const foundSection = this.map.sections.find(section => section.id === id)
+        if (!foundSection) {
+            return []
+        }
+        const options = foundSection.floors.map(floor => ({ id: floor.id, label: floor.name }))
+        return options
+    }
+    
+}
+
+
+class MapEdit extends MapView {
+
+    constructor (map: Maybe<BuildingMap>, updateMap: Maybe<(map: BuildingMap) => void>) {
+        super(map)
+        this.updateMap = updateMap
```

Comment:
```suggestion
        this.updateMap = updateMap
```

redurant command, all constructor arguments dirrectly apllying to 'this' with same name

https://www.typescriptlang.org/play?#code/MYGwhgzhAEAKBOB7ADtA3gKGt6xEDsIAXeAV2CMXgApl4BLANzCIFNoAze+YgLmIb4A5gBpodJi3YRWefABN+JesICU6LDi1EAFvQgA6LjyLQAvJ27FN2AL4ZbQA

Reply:
Done